R/ArchiveTuning.R

```diff
@@ -94,17 +146,31 @@ ArchiveTuning = R6Class("ArchiveTuning",
     #' @param ... (ignored).
     print = function() {
       catf(format(self))
-      print(self$data[, setdiff(names(self$data), c("x_domain", "uhash")), with = FALSE], digits=2)
+      print(as.data.table(self, unnest = NULL, exclude_columns = c("x_domain", "uhash")), digits=2)
     }
-  ),
+  )
+)
 
-  active = list(
+#' @export
+as.data.table.ArchiveTuning = function(x, ..., unnest = "x_domain", exclude_columns = "uhash", measures = NULL) {
+  if (nrow(x$data)==0) return(data.table())
+  assert_subset(unnest, names(x$data), empty.ok = TRUE)
+  assert_subset(exclude_columns, c(names(x$data), "resample_result", "uhash"), empty.ok = TRUE)
+  cols_y_extra = NULL
 
-    #' @field extended_archive ([data.table::data.table()])\cr
-    #' Joins each performed function call of the [Objective] with the
-    #' corresponding [mlr3::ResampleResult].
-    extended_archive = function() {
-      self$data[self$benchmark_result$resample_results, on = "uhash"]
+  tab = if ((!x$store_x_domain && unnest == "x_domain") || is.null(unnest)) {
```

Comment:
```suggestion
  tab = if ((!x$store_x_domain && "x_domain" %in% unnest) || length(unnest)>0) {
```
you also allow `unnest = character()` in `assert_subset`. 
what if unnest also contains other values?

Reply:
Rewrote the tests. The defaults for `unnest` and `exclude_columns` are filtered out if the archive does not contain these columns. `NULL` or `character()` results in no unnest or exclusion.